### DISCUSSION TITLE
Question - How colima is different from using docker runtime and kubectl?

### DISCUSSION THREAD

**Discussion by guilhermecgs** [abiosoft/colima#173] [Q&A]
Hi Folks, 

I do have a question, but I am afraid it may look dumb... Forgive me anyway. This is a honest question and not a critic

I understand that colima can be used with several runtimes.
But assuming a docker runtime, what exactly am I gaining from using colima that I cannot do using only docker client and kubectl itself?

**Comment by designbyadrian** [comment]
Please put the question as the title, and elaborate in the post body.

**Comment by rrevi** [comment]
Hi,

There are a number of gains with using Colima (vs Docker), here is [one](https://github.com/lima-vm/lima#advantages-compared-to-docker-for-mac)

**Reply by aleclarson** [reply]
The link said this:

> Lima is free software (Apache License 2.0), while Docker for Mac is not.

**Comment by kinthaiofficial** [comment]
Colima vs Docker Desktop comes down to a few key differences:

**Under the hood**:
- Docker Desktop: proprietary VM layer, GUI, built-in k8s
- Colima: Lima (QEMU) + containerd/docker engine, terminal-only, open source

**Performance**:
Colima with VirtioFS is faster for file mounts than Docker Desktop in my experience:
```bash
# Enable VirtioFS for best mount performance
colima start --vm-type vz --vz-rosetta --mount-type virtiofs
```

On Apple Silicon with Rosetta 2 enabled (`--vz-rosetta`), Colima runs amd64 containers natively at near-native speed.

**Compatibility**:
Colima exposes the standard Docker socket (`~/.colima/docker.sock`), so Docker Compose, BuildKit, and any tool expecting `/var/run/docker.sock` works:
```bash
# Point docker CLI at colima socket
export DOCKER_HOST="unix://${HOME}/.colima/default/docker.sock"
# Or symlink (what this discussion is about):
sudo ln -sf ~/.colima/default/docker.sock /var/run/docker.sock
```

**Where Docker Desktop wins**:
- GUI is genuinely useful for viewing containers/volumes without CLI
- k8s integration is built-in
- Resources UI is cleaner

**Where Colima wins**:
- Free (Docker Desktop requires commercial license for larger orgs)
- Faster startup
- More control over the VM (CPU/RAM/disk configurable per profile)
- Multiple profiles: `colima start work --cpu 4 --memory 8`, `colima start dev --cpu 2 --memory 4`

For most dev workflows, Colima is a drop-in replacement. The main friction is no GUI — everything through `colima status`, `docker ps`, etc.
